pkg/objstore/s3/s3.go

```diff
@@ -161,36 +161,57 @@ func NewBucket(logger log.Logger, conf []byte, component string) (*Bucket, error
 	return NewBucketWithConfig(logger, config, component)
 }
 
+type overrideSignerType struct {
+	signerType credentials.SignatureType
+	upstream   credentials.Provider
```

Comment:
```suggestion
	wrapped   credentials.Provider
```

It's just opinion, but we use `wrapped` for this kind of wrapped vars and overall wrapper patterns (:

Reply:
Rebase on embedding makes this unnecessary